more_itertools/more.py

```diff
@@ -3552,3 +3554,39 @@ def nth_permutation(iterable, r, index):
             break
 
     return tuple(map(pool.pop, result))
+
+
+def value_chain(*args):
+    """Yield all arguments passed to the function in the same
+    order in which they were passed. If an argument itself is
+    iterable then iterate over its values.
+
+        >>> list(value_chain(1, 2, 'foo'))
+        [1, 2, 'foo']
+        >>> list(value_chain(b'bar', [1, 2, 3], 4, {'key': 1}))
+        [b'bar', 1, 2, 3, 4, 'key']
+        >>> list(value_chain([], {}, '', 0, None))
+        [[], {}, '', 0, None]
+        >>> list(value_chain())
+        []
+
+    Note that :func:`value_chain` is the more general version
+    of :func:`prepend`:
+
+        >>> list(prepend(0, [1, 2, 3]))
+        [0, 1, 2, 3]
+        >>> list(value_chain(0, [1, 2, 3]))
+        [0, 1, 2, 3]
+
+    """
+    for value in args:
+        if isinstance(value, (str, bytes,)):
```

Comment:
We should probably add the other scalar types here, too: `int`, `float`, `bool`.

Reply:
Commit https://github.com/more-itertools/more-itertools/pull/475/commits/0d67cde0543810d8ca75005af2d9c1e94a0c7dd9.